dask/base.py

```diff
@@ -970,13 +974,15 @@ def normalize_function(func: Callable) -> Callable:
                 if len(function_cache) >= 500:
                     for k in list(function_cache)[::2]:
                         del function_cache[k]
-        function_cache[func] = result
+        function_cache[func] = result  # type: ignore
         return result
     except TypeError:  # not hashable
         return _normalize_function(func)
 
 
-def _normalize_function(func: Callable) -> Callable:
+def _normalize_function(
+    func: Callable,
+) -> Callable | Sequence[Callable] | Sequence[Any]:
```

Comment:
As a general rule,
- be as generic as possible for parameters
- be as specific as possible for return types

``Sequence[Callable] | Sequence[Any]`` is the same as ``Sequence[Any]``.

I can't see any use case where the output is a Callable.

I see below that one of the possible return types is a ``str | bytes``. While ``str | bytes`` is _technically_ a ``Sequence[Any]``, it's something that should be called out explicitly.
```suggestion
) -> tuple | str | bytes:
```

Reply:
Thanks this set me on the right path (I believe).